src/NuGet.Services.Metadata.Catalog.Monitoring/Resources/PackageTimestampMetadataResourceV3.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using NuGet.Protocol;
+using NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test;
+using Microsoft.Extensions.Logging;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring.Providers
+{
+    public class PackageTimestampMetadataResourceV3 : IPackageTimestampMetadataResource
+    {
+        public PackageTimestampMetadataResourceV3(
+            RegistrationResourceV3 registration, 
+            ILogger<PackageTimestampMetadataResourceV3> logger)
+        {
+            _registration = registration;
+        }
+        
+        private readonly RegistrationResourceV3 _registration;
+        private readonly ILogger<PackageTimestampMetadataResourceV3> _logger;
+
+        /// <summary>
+        /// Parses the registration for the package specified by the <see cref="ValidationContext"/> and returns a <see cref="PackageTimestampMetadata"/> created from the package's catalog entry.
+        /// If the registration page is missing, search for a deletion catalog entry and use its timestamp.
+        /// </summary>
+        public async Task<PackageTimestampMetadata> Get(ValidationContext data)
+        {
+            var registrationPage = await _registration.GetPackageMetadata(data.Package, _logger.AsCommon(), data.CancellationToken);
```

Comment:
was originally needed but not anymore--will delete the `PackageTimestampMetadataResourceV3` related files